File: CS/MVVMDemo.Wpf/Modules/Commands/AutoCommandsChangeNotificationsViewModel.cs
using DevExpress.Mvvm;
using DevExpress.Mvvm.DataAnnotations;
using System.Windows;

namespace MVVMDemo.Commands {
    public class AutoCommandsChangeNotificationsViewModel : ViewModelBase {
        [Command(UseCommandManager = false)]
        public void Save() {
            MessageBox.Show("Action: Save");
        }
        public bool CanSave() {
            return CanExecuteSaveCommand;
        }

        public bool CanExecuteSaveCommand {
            get { return GetProperty(() => CanExecuteSaveCommand); }
            set {
                #region !
                if(SetProperty(() => CanExecuteSaveCommand, value))
                    RaiseCanExecuteChanged(() => Save());
                #endregion            }
            }
        }

        public AutoCommandsChangeNotificationsViewModel() {
            CanExecuteSaveCommand = true;
        }
    }
}

File: VB/MVVMDemo.Wpf/Modules/Commands/AutoCommandsChangeNotificationsViewModel.vb
Imports DevExpress.Mvvm
Imports DevExpress.Mvvm.DataAnnotations
Imports System.Windows

Namespace MVVMDemo.Commands
    Public Class AutoCommandsChangeNotificationsViewModel
        Inherits ViewModelBase

        <Command(UseCommandManager := False)>
        Public Sub Save()
            MessageBox.Show("Action: Save")
        End Sub
        Public Function CanSave() As Boolean
            Return CanExecuteSaveCommand
        End Function

        Public Property CanExecuteSaveCommand() As Boolean
            Get
                Return GetProperty(Function() CanExecuteSaveCommand)
            End Get
            Set(ByVal value As Boolean)
'                #Region "!"
                If SetProperty(Function() CanExecuteSaveCommand, value) Then
                    RaiseCanExecuteChanged(Sub() Save())
                End If
'                #End Region '            }
            End Set
        End Property

        Public Sub New()
            CanExecuteSaveCommand = True
        End Sub
    End Class
End Namespace
